src/service/document-click.js

```diff
@@ -133,8 +148,16 @@ export class ClickHandler {
     if (!target || !target.href) {
       return;
     }
-    Services.urlReplacementsForDoc(target).maybeExpandLink(target);
 
+    // First check if need to handle external link decoration.
+    let defaultExpandParamsUrl = null;
+    if (this.appendExtraParams_ && !this.isEmbed_) {
+      // Only decorate outgoing link when needed to and is not in FIE.
+      defaultExpandParamsUrl = getExtraParamsUrl(this.ampdoc.win, target);//this.getExternalNavDecoration_(target);
```

Comment:
Would it be simpler to do all this in maybeExpandLink?

Reply:
Sure that will work. Basically I pass a boolean to `maybeExpandLink`, and if it's set to true, I call `getExtraParamsUrl` in `maybeExpandLink` function. 

I don't see that much simpler though. I'll leave it as it is for now : ) Please let me know if I am missing anything here?

Reply:
nit: can rm //this.getExternalNavDecoration_(target);